### PR TITLE
LF: Drop typeparameter from KeyWithMaintainers

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Following up #10827 and #10921, we drop type parameter to
KeyWithMaintainers, and use the `Versioned` wrapper introduced in

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala [hunk 1]
```diff
@@ -220,32 +211,23 @@ object Node {
...
+  final case class KeyWithMaintainers(key: Value, maintainers: Set[Party])
+      extends CidContainer[KeyWithMaintainers] {
 
-    def map[Val2](f: Val => Val2): KeyWithMaintainers[Val2] =
+    def map(f: Value => Value): KeyWithMaintainers =
       copy(key = f(key))
-  }
 
-  object KeyWithMaintainers {
-    implicit class CidContainerInstance[Val <: CidContainer[Val]](key: KeyWithMaintainers[Val])
```
Comment (cocreature) [inline daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala:234]: :fire:

**Review by cocreature — APPROVED** [review]
Thanks!

**Comment by remyhaemmerle-da** [comment]
An other `Versioned` change for you @oliverse-da

**Comment by oliverse-da** [comment]
> An other `Versioned` change for you @oliverse-da

Thanks for the heads up, @remyhaemmerle-da , this one was quite simple to merge into canton.

**Pull request merged**